src/mockups/UniswapV2Router02Mock.sol

```diff
@@ -0,0 +1,38 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity >=0.4.22 <0.9.0;
+
+import "../interfaces/IERC20.sol";
+
+import "../../lib/forge-std/src/Test.sol";
+
+contract UniswapV2Router02Mock is Test {
+    using stdStorage for StdStorage;
+
+    function swapExactTokensForTokens(
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external virtual returns (uint256[] memory amounts) {
+        //Cheat balance of
+        uint256 slot = stdstore.target(address(path[0])).sig(IERC20(path[0]).balanceOf.selector).with_key(
+            address(msg.sender)
+        ).find();
+        bytes32 loc = bytes32(slot);
+        bytes32 mockedBalanceOf = bytes32(abi.encode(IERC20(path[0]).balanceOf(msg.sender) - amountIn));
+        vm.store(address(path[0]), loc, mockedBalanceOf);
+
+        //Cheat balance of
+        uint256 slot2 =
+            stdstore.target(address(path[1])).sig(IERC20(path[1]).balanceOf.selector).with_key(address(to)).find();
+        bytes32 loc2 = bytes32(slot2);
+        bytes32 mockedBalanceOf2 = bytes32(abi.encode(amountOutMin));
```

Comment:
end balance should not be amountOut but current balance increased with amountOutMin (and probably more). In your current implementation, since `to` is the vaultaddress, it sets the balance of the vault address of that token to amountOutMin, which would screw up balances if your vault already had some of the output token

Reply:
changed this to withdraw to action instead that way we don't have to check. The original preBalance check was to check this btw.